t/api/05-machines.t

```diff
@@ -87,6 +87,18 @@ $t->post_ok('/api/v1/machines',
     form => {name => "testmachine", backend => "qemu", "settings[TEST]" => "val1", "settings[TEST2]" => "val1"})
   ->status_is(200);
 my $machine_id = $t->tx->res->json->{id};
+is_deeply(
+    OpenQA::Test::Case::find_most_recent_event($t->app->schema, 'table_create'),
+    {
+        id          => $machine_id,
+        table       => 'Machines',
+        name        => "testmachine",
+        backend     => "qemu",
+        settings    => [{key => 'TEST', value => 'val1'}, {key => 'TEST2', value => 'val1'}],
```

Comment:
I had a brief look at `sub _prepare_settings` and the settings are not sorted. That means this check is relying on the order of hash keys which might be random. The same counts for the other checks.

Reply:
as an alternative to sorting the output don't use `is_deeply` but rather check that individual expected values are present. I do not see the reason why this check has to fail if an additional setting would be added.

Reply:
Yes, the overuse of `is_deeply` can make maintaining tests harder and it also feels wrong to introduce sorting just for the sake of tests.

Reply:
It seemed appropriate since we literally don't specify or test anywhere else what is supposed to be contained in the event. This is not a case of 1000 calls with redundant checks.